internal/common/rules/globalrulescache.go

```diff
@@ -0,0 +1,16 @@
+package rules
+
+import "gonum.org/v1/gonum/mat"
+
+var AvailableRules = map[string]RuleMatrix{}
+
+// RegisterNewRule Creates and registers new rule based on inputs
+func RegisterNewRule(ruleName string, requiredVariables []string, applicableMatrix mat.Dense, auxiliaryVector mat.VecDense) *RuleMatrix {
+	if _, ok := AvailableRules[ruleName]; ok {
+		panic("attempt to re-register rule with name " + ruleName)
```

Comment:
We should use an error code rather than a panic because a bad client shouldn't be able to crash the simulation

Reply:
agreed - if clients can call this we should error instead of panic

Reply:
Yup agreed and changed